src/main/kotlin/app/Configurator.kt

```diff
@@ -0,0 +1,178 @@
+            if (merged.silent == null) {
+                merged.silent = level.silent
+            }
+        }
+
+        return merged
+    }
+
+    // Loads config file from specified path.
+    private fun loadConfig(path: String): Options {
+        val options = Options()
+
+        try {
+            val file = File(path, configFileName)
+
+            if (!file.exists() || !file.isFile) {
+                return options  // No configuration file
+            }
+
+            val config = PropertiesConfiguration(file)
+
+            // Accessing configuration properties. Unknown values should be
+            // null to distinguish them from specified values from other levels.
+            options.username = config.getString(configUsername, null)
+            options.password = config.getString(configPassword, null)
+            options.silent = config.getBoolean(configSilent, null)
+        }
+        catch (e: ConfigurationException) {
+            // Error while loading the properties file.
+        }
+        catch (e: SecurityException) {
+            // Read access denied.
+        }
+
+        return options
+    }
+
+    // Saves config file to specified path. Returns true on success.
+    private fun saveConfig(path: String, options: Options): Boolean {
+        try {
+            val file = File(path, configFileName)
+
+            file.createNewFile()  // Creates a new file if it didn't exist.
+
+            val config = PropertiesConfiguration(file)
+
+            if (options.username != null) {
+                config.setProperty(configUsername, options.username)
+            }
+            if (options.password != null) {
+                config.setProperty(configPassword, options.password)
+            }
+            if (options.silent != null) {
+                config.setProperty(configSilent, options.silent)
+            }
+
+            config.save(file)
+
+            return true
+        }
+        catch (e: IOException) {
+            // IO error occurred.
+        }
+        catch (e: SecurityException) {
+            // Read access denied.
+        }
+        catch (e: ConfigurationException) {
+            // Error while loading or saving the properties file.
+        }
+
+        return false
+    }
+
+    fun setCurrentOptions(options: Options) {
+        current = options
+    }
+
+    fun loadUserLevelConfig() {
+        if (userDir != null) {
+            config = loadConfig(userDir)
+        }
+    }
+
+    fun saveUserLevelConfig(options: Options): Boolean {
+        if (userDir != null) {
+            return saveConfig(userDir, options)
+        }
+
+        return false
+    }
+
+    fun createOptions(pair: List<String>): Options {
+        val options = Options()
+
+        if (pair.count() != 2) {
+            return options
+        }
+
+        val (key, value) = pair
+
+        when (key) {
+            configUsername -> {
+                if (UsernameValidator().isValidUsername(value)) {
```

Comment:
I really think we should have @osilla define user UI. Ideally, I would want them to launch sourcerer, get asked what their user info is, etc, and have that info saved to a config file. If they ever need to change it, they can always delete it from the config file. My worry here is that by building a complicated options/commands system, we will make it necessary to read documentation before they can use it. Of course, we can always provide a wrapper bash for this, so maybe it's the way to go, but I would like @osilla to have authority on UI because it's how we make UI consistent.